src/lex/mod.rs

```diff
@@ -424,19 +426,32 @@ impl Lexer {
                     Ok(match c {
                         // escaped newline: "a\
                         // b"
-                        b'\n' => return self.parse_single_char(string),
+                        b'\n' => unreachable!("should be handled earlier"),
```

Comment:
This is a holdover from when I changed backslashes to be handled earlier: https://github.com/jyn514/rcc/blob/519345079a25f6b6f9b2b267893243c29cb1788e/src/lex/mod.rs#L86